planner/planner.go

```diff
@@ -280,8 +283,52 @@ func (p *MyPlanner) cityStatsHandler(context *gin.Context) {
 	context.JSON(http.StatusOK, view)
 }
 
-func (p *MyPlanner) Planning(ctx context.Context, planningRequest *PlanningReq, user string) (resp PlanningResponse) {
-	planningResponse := p.Solver.Solve(ctx, planningRequest)
+func (p *MyPlanner) Planning(ctx context.Context, planningRequest PlanningReq, user string, withNearbyCities bool) (resp PlanningResponse) {
+	logger := iowrappers.Logger
+	var planningResponse *PlanningResp
+	if withNearbyCities {
+		lat, lng, err := p.Solver.Searcher.Geocode(ctx, &iowrappers.GeocodeQuery{
+			City:              planningRequest.Location.City,
+			AdminAreaLevelOne: planningRequest.Location.AdminAreaLevelOne,
+			Country:           planningRequest.Location.Country,
+		})
+		if err != nil {
+			return PlanningResponse{Err: err}
+		}
+		logger.Debugf("->Planning: lat, lng from Geocode: %.4f, %.4f", lat, lng)
+
+		// convert km to m for nearbyCityResponse search query
+		nearbyCityResponse, err := p.Solver.Searcher.NearbyCities(ctx,
+			&iowrappers.NearbyCityRequest{
+				ApiKey: p.GeonamesApiKey,
+				Location: POI.Location{
+					Latitude:          lat,
+					Longitude:         lng,
+					City:              planningRequest.Location.City,
+					AdminAreaLevelOne: planningRequest.Location.AdminAreaLevelOne,
+					Country:           planningRequest.Location.Country,
+				},
+				Radius: float64(planningRequest.SearchRadius / 1000),
+				Filter: go_geonames.CityWithPopulationGreaterThan15000,
+			})
+		if err != nil {
+			return PlanningResponse{Err: err}
+		}
+		locations := MapSlice[iowrappers.City, POI.Location](nearbyCityResponse.Cities, toLocation)
```

Comment:
How many nearby cities do we usually get? To many nearby cities, can increase the load of the server if the traffic gets big. We can consider adding a cap to the number of parallel search requests.

Reply:
Good point. Actually in design doc we proposed a limit. Any number in mind?

Reply:
We can go with 10 first, and set this cap as a configurable parameter and adjust later. Do you know if Heroku automatically increase the number of servers we are using if the computation load increases?

Reply:
I don't think Heroku does that. If we deploy on kubernetes it will do that.